src/main/java/com/atlassian/jira/cloud/jenkins/config/JiraCloudSiteConfig2.java

```diff
@@ -0,0 +1,188 @@
+package com.atlassian.jira.cloud.jenkins.config;
+
+import com.atlassian.jira.cloud.jenkins.auth.AccessTokenRetriever;
+import com.atlassian.jira.cloud.jenkins.common.model.AppCredential;
+import com.atlassian.jira.cloud.jenkins.tenantinfo.CloudIdResolver;
+import com.atlassian.jira.cloud.jenkins.util.SecretRetriever;
+import com.atlassian.jira.cloud.jenkins.util.SiteValidator;
+import com.atlassian.jira.cloud.jenkins.util.WebhookUrlValidator;
+import com.cloudbees.plugins.credentials.CredentialsMatchers;
+import com.cloudbees.plugins.credentials.common.StandardListBoxModel;
+import com.cloudbees.plugins.credentials.domains.URIRequirementBuilder;
+import hudson.Extension;
+import hudson.model.AbstractDescribableImpl;
+import hudson.model.Descriptor;
+import hudson.security.ACL;
+import hudson.util.FormValidation;
+import hudson.util.ListBoxModel;
+import jenkins.model.Jenkins;
+import org.apache.commons.lang.StringUtils;
+import org.jenkinsci.plugins.plaincredentials.StringCredentials;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.DoNotUse;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.QueryParameter;
+import org.kohsuke.stapler.interceptor.RequirePOST;
+
+import javax.inject.Inject;
+import java.util.Objects;
+import java.util.Optional;
+
+import static com.atlassian.jira.cloud.jenkins.Config.ATLASSIAN_API_URL;
+import static java.util.Objects.requireNonNull;
+
+/**
+ * This class encapsulates Jira Cloud site configuration to be used to send build information to
+ * Jira.
+ */
+public class JiraCloudSiteConfig2 extends AbstractDescribableImpl<JiraCloudSiteConfig2> {
```

Comment:
Copy of `JiraCloudSiteConfig`, but modified to support a webhook URL instead of a Client ID.